ui/src/app/base/components/TableMenu/TableMenu.tsx

```diff
@@ -38,7 +38,7 @@ const TableMenu = <L extends null>({
       // See: https://github.com/canonical-web-and-design/react-components/issues/377
       positionNode={positionNode || undefined}
       toggleAppearance="base"
-      toggleClassName="u-no-margin--bottom p-table-menu__toggle"
+      toggleClassName="u-no-margin--bottom p-table-menu__toggle is-dense"
```

Comment:
I'm not sure about adding this as a drive-by fix. We use `TableMenu` in quite a few places so there's potential for this to have an unintended side effect *somewhere*. Plus it kinda makes the button oblong shaped instead of square.

In other places we just make the dropdown disabled if no actions are available, which should address any jumping around.

Reply:
Good point, reverted.